static/src/javascripts/projects/commercial/modules/creatives/fabric-v1.js

```diff
@@ -0,0 +1,212 @@
+            }
+
+            // #? `classList.add` takes multiple arguments, but we are using it
+            // here with arity 1 because polyfill.io has incorrect support with IE 10 and 11.
+            // One may revert to this.adSlot.classList.add('ad-slot--fabric-v1', 'ad-slot--fabric', 'content__mobile-full-width');
+            // When support is correct or when we stop supporting IE <= 11
+            this.adSlot.classList.add('ad-slot--fabric-v1');
+            this.adSlot.classList.add('ad-slot--fabric');
+            this.adSlot.classList.add('content__mobile-full-width');
+
+            if (
+                this.adSlot.parentNode &&
+                this.adSlot.parentNode instanceof HTMLElement &&
+                this.adSlot.parentNode.classList.contains(
+                    'top-banner-ad-container'
+                )
+            ) {
+                this.adSlot.parentNode.classList.add(
+                    'top-banner-ad-container--fabric'
+                );
+            }
+
+            if (this.params.viewabilityTracker) {
+                addViewabilityTracker(
+                    this.adSlot,
+                    this.params.id,
+                    this.params.viewabilityTracker
+                );
+            }
+
+            return true;
+        });
+    }
+
+    updateBgPosition() {
+        if (this.scrollType === 'parallax') {
+            const scrollAmount =
+                Math.ceil(this.adSlot.getBoundingClientRect().top * 0.3) + 20;
+            fastdom.write(() => {
+                if (this.scrollingBg) {
+                    this.scrollingBg.style.backgroundPosition = `50% ${scrollAmount}%`;
+                    this.scrollingBg.classList.add('ad-scrolling-bg-parallax');
+                }
+            });
+        } else if (this.scrollType === 'fixed' && !hasBackgroundFixedSupport) {
+            const adRect = this.adSlot.getBoundingClientRect();
+            const vPos =
+                (window.innerHeight - adRect.bottom + adRect.height / 2) /
+                window.innerHeight *
+                100;
+            fastdom.write(() => {
+                if (this.scrollingBg) {
+                    this.scrollingBg.style.backgroundPosition = `50% ${vPos}%`;
+                }
+            });
+        }
+        this.layer2Animation();
+    }
+
+    layer2Animation() {
+        let inViewB;
+        if (
+            this.params.layerTwoAnimation === 'enabled' &&
+            isEnhanced &&
+            !isIE10OrLess
+        ) {
+            inViewB =
+                detect.getViewport().height >
+                this.adSlot.getBoundingClientRect().top;
+            fastdom.write(() => {
+                if (this.layer2) {
+                    this.layer2.classList
+                        .add(`ad-scrolling-text-hide${this.params.layerTwoAnimationPosition ? `-${this.params.layerTwoAnimationPosition}` : ''}`);
+                }
+                if (this.layer2 && inViewB) {
```

Comment:
We need to guard `this.layer2` again, since Flow believes that the previous call to `this.layer2.classList.add` could have altered `this.layer2`, as far as I can tell.